duffel_api/models/order_change_offer.py

```diff
@@ -28,50 +156,31 @@ class OrderChangeOffer:
     class InvalidRefundType(Exception):
         """Invalid refund type provided"""
 
-    def __init__(self, json):
-        for key in json:
-            value = maybe_parse_date_entries(key, json[key])
-            if (
-                key == "refund_to"
-                and value not in OrderChangeOffer.allowed_refund_types
-            ):
-                raise OrderChangeOffer.InvalidRefundType(value)
-            if key == "slices":
-                value = OrderChangeOfferSlices(value)
-            setattr(self, key, value)
-
-
-class OrderChangeOfferSlices:
-    """The slices to be added and/or removed"""
-
-    def __init__(self, json):
-        add = list(map(lambda add: OrderChangeOfferSlicesAdd(add), json["add"]))
-        remove = list(
-            map(lambda remove: OrderChangeOfferSlicesRemove(remove), json["remove"])
+    @classmethod
+    def from_json(cls, json: dict):
+        """Construct a class instance from a JSON response."""
+        return cls(
+            id=json["id"],
+            order_change_id=json["order_change_id"],
+            live_mode=json["live_mode"],
+            change_total_amount=json["change_total_amount"],
+            change_total_currency=json.get("change_total_currency"),
+            new_total_amount=json.get("new_total_amount"),
+            new_total_currency=json["new_total_currency"],
+            penalty_amount=json.get("penalty_amount"),
+            penalty_currency=json.get("penalty_currency"),
+            refund_to=json["refund_to"],
+            created_at=datetime.strptime(json["created_at"], "%Y-%m-%dT%H:%M:%S.%fZ"),
+            updated_at=datetime.strptime(json["updated_at"], "%Y-%m-%dT%H:%M:%S.%fZ"),
+            expires_at=parse_datetime(json["expires_at"]),
+            slices=OrderChangeOfferSlices.from_json(json["slices"]),
         )
 
-        setattr(self, "add", add)
-        setattr(self, "remove", remove)
-
-
-class OrderChangeOfferSlicesAdd:
-    """The slice to be added"""
-
-    allowed_place_types = ["airport", "city"]
-
-    class InvalidPlaceType(Exception):
-        """Invalid type of place"""
-
-    def __init__(self, json):
-        for key in json:
-            value = maybe_parse_date_entries(key, json[key])
-            if key in ["destination", "origin"]:
-                value = Place(value)
-            elif key in ["destination_type", "origin_type"]:
-                if value not in OrderChangeOfferSlicesAdd.allowed_place_types:
-                    raise OrderChangeOfferSlicesAdd.InvalidPlaceType(value)
-            # TODO(nlopes): maybe convert duration to a timedelta or Duration
-            setattr(self, key, value)
-
 
-OrderChangeOfferSlicesRemove = OrderChangeOfferSlicesAdd
+def parse_datetime(value: str) -> datetime:
```

Comment:
See [comment above](https://github.com/duffelhq/duffel-api-python/pull/79/files#r784900678) about defining this in utils and using everywhere!